logical/logical.go

```diff
@@ -2,6 +2,18 @@ package logical
 
 import log "github.com/mgutz/logxi/v1"
 
+// BackendType is the type of backend that is being implemented
+type BackendType string
+
+// The these are the types of backends that can be derived from
+// logical.Backend
+const (
+	TypeUnknown     BackendType = "unknown"
+	TypeLogical     BackendType = "logical"
```

Comment:
These string values are shown to users and show up on the log as well.

CLI:
```
$ vault auth-enable -plugin-name=passthrough-plugin plugin
Error: Error making API request.

URL: POST http://127.0.0.1:8200/v1/sys/auth/passthrough-plugin
Code: 400. Errors:

* cannot mount 'passthrough-plugin' of type 'passthrough' as an auth backend
```

Logs:
```
2017/07/06 13:21:01.202610 [ERROR] sys: enable auth mount failed: path=mock-plugin/ error=cannot mount 'mock-plugin' of type 'unknown' as an auth backend
```

Should I keep the type names to `TypeLogical` `TypeCredential`, etc. or should those be changed as well?